lib/ecto/changeset.ex

```diff
@@ -729,6 +729,29 @@ defmodule Ecto.Changeset do
   end
 
   @doc """
+  Validates a change, of type enum, is a subset of the given enumerable. Like
+  validate_inclusion for lists.
+
+  ## Options
+
+    * `:message` - the message on failure, defaults to "\#{x} is invalid"
+
+  ## Examples
+
+      validate_subset(changeset, :pets, ["cat", "dog", "parrot"])
+      validate_subset(changeset, :lottery_numbers, 0..99)
+
+  """
+  @spec validate_subset(t, atom, Enum.t) :: t
+  def validate_subset(changeset, field, data, opts \\ []) do
+    validate_change changeset, field, {:subset, data}, fn _, values ->
+      Enum.flat_map values, fn(x) ->
```

Comment:
I am not sure. :( I am thinking we would be fine with a single validation error because if there is a wrong entry, it is very likely an error in the UI (or the user is fiddling with things he is not supposed to). Thoughts?